src/posix/platform/netif.cpp

```diff
@@ -379,6 +793,284 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
+                        }
+                    }
+                }
+            }
+            else
+            {
+                otNetifMulticastAddress netAddr;
+                netAddr.mAddress = addr;
+
+                error = otIp6SubscribeMulticastAddress(aInstance, &addr);
+                if ((error == OT_ERROR_NONE) || (error == OT_ERROR_ALREADY))
+                {
+                    otLogNotePlat("ADD [M] %s%s",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  error == OT_ERROR_ALREADY ? " (already subscribed, ignored)" : "");
+                    error = OT_ERROR_NONE;
+                }
+                else
+                {
+                    otLogWarnPlat("ADD [M] %s failed (%s)",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  otThreadErrorToString(error));
+                }
+                SuccessOrExit(error);
+            }
+        }
+        else if (rtm->rtm_type == RTM_DELADDR
+#ifdef RTM_DELMADDR
+                 || rtm->rtm_type == RTM_DELMADDR
+#endif
+        )
+        {
+            if (!addr.IsMulticast())
+            {
+                error = otIp6RemoveUnicastAddress(aInstance, &addr);
+                if ((error == OT_ERROR_NONE) || (error == OT_ERROR_NOT_FOUND))
+                {
+                    otLogNotePlat("DEL [U] %s%s",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
+
+                    error = OT_ERROR_NONE;
+                }
+                else
+                {
+                    otLogWarnPlat("DEL [U] %s failure (%d)",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  error);
+                }
+            }
+            else
+            {
+                error = otIp6UnsubscribeMulticastAddress(aInstance, &addr);
+                if ((error == OT_ERROR_NONE) || (error == OT_ERROR_NOT_FOUND))
+                {
+                    otLogNotePlat("DEL [M] %s%s",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
+                    error = OT_ERROR_NONE;
+                }
+                else
+                {
+                    otLogWarnPlat("DEL [M] %s failure (%d)",
+                                  inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                  error);
+                }
+            }
+        }
+    }
+
+exit:;
```

Comment:
```
exit:
    return;
```
Or we may do the logging here to avoid duplicate code?

Reply:
I consolidated a lot of the logging -- I think it's substantially better now.